executor/aggregate.go

```diff
@@ -862,6 +871,56 @@ func (e *HashAggExec) getPartialResults(groupKey string) []aggfuncs.PartialResul
 	return partialResults
 }
 
+func (e *HashAggExec) executeFirstRow(ctx context.Context, chk *chunk.Chunk) (err error) {
+	for {
+		if e.firstRowProcessed == e.childResult.NumRows() {
+			mSize := e.childResult.MemoryUsage()
+			e.firstRowProcessed = 0
+			err := Next(ctx, e.children[0], e.childResult)
+			e.memTracker.Consume(e.childResult.MemoryUsage() - mSize)
+			if err != nil {
+				return err
+			}
+			// no more data.
+			if e.childResult.NumRows() == 0 {
+				return nil
+			}
+			e.groupKeyBuffer, err = getGroupKey(e.ctx, e.childResult, e.groupKeyBuffer, e.GroupByItems)
+			if err != nil {
+				return err
+			}
+		}
+
+		failpoint.Inject("unparallelHashAggError", func(val failpoint.Value) {
+			if val.(bool) {
+				failpoint.Return(errors.New("HashAggExec.unparallelExec error"))
+			}
+		})
```

Comment:
Please add some test cases to cover the failpoint.

Reply:
I have added a test case for it.